### ISSUE TITLE
Rancher-monitoring: query issue on daemonset

### ISSUE THREAD

**Issue by spencergilbert** [rancher/rancher#20162]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Enable rancher monitoring, deploy a daemonset, view daemonset

**Result:**
cpu, memory, memory usage, cpu usage - all appear to be inaccurate, some are double the actual amount

**Other details that may be helpful:**
Looks like it should be resolved by including `container_name!="POD", container_name!="", ` in the prometheus query - this happens on other dashboards but not here

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.2
- Installation option (single install/HA): HA

**Comment by spencergilbert** [comment]
CPU: 
`sum(rate(container_cpu_usage_seconds_total{container_name!="POD", container_name!="", namespace="$daemonset_namespace",pod_name=~"$daemonset_name.*"}[5m]))`

Memory:
`sum(container_memory_working_set_bytes{namespace="$daemonset_namespace",pod_name=~"$daemonset_name.*", container_name!="POD", container_name!=""})`

CPU Usage:
`sum by (pod_name) (rate(container_cpu_usage_seconds_total{namespace="$daemonset_namespace",pod_name=~"$daemonset_name.*", container_name!="POD", container_name!=""}[5m]))`

Memory Usage
`sum by (pod_name) (container_memory_working_set_bytes{namespace="$daemonset_namespace",pod_name=~"$daemonset_name.*", container_name!="POD", container_name!=""})`

This is the updated dashboard json from grafana after I made these updates - happy to PR the system-charts repo if this looks correct
[updated-json.txt](https://github.com/rancher/rancher/files/3167962/updated-json.txt)

**Comment by spencergilbert** [comment]
Seeing something similar on statefulset dashboard with cpu and memory

**Comment by davidnuzik** [comment]
I too am observing this. It's actually fairly noticeable if you compare rancher charts to the grafana ones grafana is showing 2x in most cases.

**Comment by spencergilbert** [comment]
As I think about it, I wonder if the graphs embedded in the rancher ui are also affected - I'd try to verify but I'm not sure where that code is.

**Comment by thxCode** [comment]
@spencergilbert , for rancher ui, the expressions are in https://github.com/rancher/rancher/blob/master/pkg/controllers/user/monitoring/metricExpression.go. This is indeed a bug for ignoring `conatiner_name!=""` in some expressions, but I prefer to keep the counting for `POD` container, as the pause container is also a part of one workload.

**Comment by spencergilbert** [comment]
Sounds good to me, I'm still wrapping my head around the kubernetes monitoring here so I'll keep that in mind with the POD container.

Thanks!

**Comment by alena1108** [comment]
The fix is in system-charts dev branch, and can be tested with master build

**Comment by davidnuzik** [comment]
Fixed in v2.2.4-rc1 in backport issue #20242 
When I have time I will check in latest master (v2.3) in this issue as well.

**Comment by spencergilbert** [comment]
Thanks everyone!

**Comment by davidnuzik** [comment]
Latest master has been tested; I have the same findings as my tests done recently for the backport issue #20242 (v2.2.4-rc1); this is working now.

Version: Master (v2.3) (5/29/19)

The PR https://github.com/rancher/system-charts/pull/43 contains a fix and here is the overview:

>Problem: There are two kubelet scraping targets on Prometheus, one is scraping /metrics, another one is scraping /metrics/cadvisor. The metrics from /metrics endpoint are not including container_name. So the container_* expression will double the actual mount without
conatiner_name!=""

>Fix: Add `container_name!=""` into the expression

This should fix the inaccurate Grafana dashboards that were in some cases (when `container_name!=""` was not there) showing 2x the values.

I tested again and now I observe data that matches what we see in Rancher charts. Grafana is no longer showing 2x the values.

I tested by creating a CPU load on an ubuntu workload. I utilized the `stress` package to put 50% of cpu cores under load then 100% cpu cores under load. This way we have consistent cpu utilization more or less. I ensured that for each test the Rancher wheels and charts matched what we see in Grafana -- down to the Deployment level this appears accurate now (check Cluster, Node, Pod, Deployment [memory], etc and compare each to Rancher UI). I did spot checks as well for memory usage, etc which also appears accurate now.
